src/components/select/select.ts

```diff
@@ -0,0 +1,377 @@
+      activeItemIndex ?? -1,
+      -1
+    );
+    this.selectItem(this.allItems[prev], true);
+  }
+
+  protected selectFirstActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[0]);
+    const item = this.allItems[index];
+
+    if (item.value !== this.value) {
+      this.selectItem(item, true);
+    }
+  }
+
+  protected selectLastActiveItem() {
+    const items = this.allItems.filter((i) => !i.disabled);
+    const index = this.allItems.indexOf(items[items.length - 1]);
+    const item = this.allItems[index];
+
+    if (item.value !== this.value) {
+      this.selectItem(item, true);
+    }
+  }
+
+  protected selectItemByKey(event: KeyboardEvent): void {
+    if (
+      !event ||
+      !event.key ||
+      event.key.length > 1 ||
+      event.key === ' ' ||
+      event.key === 'spacebar'
+    ) {
+      // ignore longer keys ('Alt', 'ArrowDown', etc) AND spacebar (used of open/close)
+      return;
+    }
+
+    const currentTime = Date.now();
+
+    if (currentTime - this.lastKeyTime > 500) {
+      this.searchTerm = '';
+    }
+
+    this.searchTerm += event.key;
+    this.lastKeyTime = currentTime;
+
+    const item = this.allItems
+      .filter((i) => !i.disabled)
+      .find((i) =>
+        i.textContent
+          ?.trim()
+          .toLowerCase()
+          .startsWith(this.searchTerm.toLowerCase())
+      );
+    if (item) this.selectItem(item);
+  }
+
+  protected handleInputKeyboardEvents(event: KeyboardEvent) {
+    event.stopPropagation();
+    const key = event.key.toLowerCase();
+
+    if (
+      event.altKey &&
+      (key === 'arrowup' ||
+        key === 'arrowdown' ||
+        key === 'down' ||
+        key === 'up')
+    ) {
+      this.toggle();
+      return;
+    }
+
+    if (!this.open) {
+      switch (key) {
+        case 'space':
+        case 'spacebar':
+        case ' ':
+        case 'enter':
+          event.preventDefault();
+          this.show();
+          return;
+        case 'arrowdown':
+        case 'down':
+        case 'arrowright':
+        case 'right':
+          event.preventDefault();
+          this.selectNext();
+          return;
+        case 'arrowup':
+        case 'up':
+        case 'arrowleft':
+        case 'left':
+          event.preventDefault();
+          this.selectPrev();
+          return;
+        case 'home':
+          event.preventDefault();
+          this.selectFirstActiveItem();
+          return;
+        case 'end':
+          event.preventDefault();
+          this.selectLastActiveItem();
+          return;
+        default:
+          break;
+      }
+    } else if (key === 'tab' || (event.shiftKey && key === 'tab')) {
+      this.hide();
+    }
+
+    this.handleKeyDown(event);
+    this.selectItemByKey(event);
+  }
+
+  protected renderInnerSlots(el: HTMLElement, slot: string) {
+    el.setAttribute('slot', slot);
+    return html`${el.cloneNode(true)}`;
```

Comment:
Must be a way to re-project the prefix and suffix slots without cloning them :( This drops the connection the slotted elements had outside the component's shadow DOM and add limitations can be used. I realize in most cases it's a simple icon, but still.

Reply:
Assuming you can't re-project via wrapper since it'll trigger the prefixed/suffixed state of the input and mess it up?
Since the input doesn't have a default slot, perhaps we can use that and project/unproject prefix/suffix only when one is present by manipulating the wrapper slot target?
I tried something similar here: [https://lit.dev/playground](https://lit.dev/playground/#project=W3sibmFtZSI6InNsb3Qtc25pZmZlci50cyIsImNvbnRlbnQiOiJpbXBvcnQge2h0bWwsIGNzcywgTGl0RWxlbWVudH0gZnJvbSAnbGl0JztcbmltcG9ydCB7Y3VzdG9tRWxlbWVudCwgcHJvcGVydHksIHF1ZXJ5QXNzaWduZWRFbGVtZW50cywgc3RhdGUgfSBmcm9tICdsaXQvZGVjb3JhdG9ycy5qcyc7XG5cbkBjdXN0b21FbGVtZW50KCdzbG90LXNuaWZmZXInKVxuZXhwb3J0IGNsYXNzIFNsb3RTbmlmZmVyIGV4dGVuZHMgTGl0RWxlbWVudCB7XG4gIHN0YXRpYyBzdHlsZXMgPSBjc3NgcCB7IGNvbG9yOiBibHVlIH1gO1xuXG4gIEBwcm9wZXJ0eSgpXG4gIG5hbWUgPSAnU29tZWJvZHknO1xuICBcbiAgQHF1ZXJ5QXNzaWduZWRFbGVtZW50cyh7IHNsb3Q6ICduYW1lZCcgfSlcbiAgaXRlbXMhOiBBcnJheTxIVE1MRWxlbWVudD47XG4gIFxuICBAc3RhdGUoKVxuICBwcm90ZWN0ZWQgc2xvdFBvcHVsYXRlZCA9IGZhbHNlO1xuXG4gIHJlbmRlcigpIHtcbiAgICByZXR1cm4gaHRtbGBcbiAgICAgICAgPHA-JHt0aGlzLnNsb3RQb3B1bGF0ZWQgPyBgSSB3aWxsIHJlLXNsb3RgIDogYEknbSBlbXB0eSBpbnNpZGVgfSAoJHt0aGlzLml0ZW1zLmxlbmd0aH0pPC9wPlxuICAgICAgICA8c3BhbiBzbG90PVwiJHt0aGlzLnNsb3RQb3B1bGF0ZWQgPyBcInRhcmdldFwiIDogXCJcIn1cIj5cbiAgICAgICAgICA8c2xvdCBuYW1lPVwibmFtZWRcIiBAc2xvdGNoYW5nZT0ke3RoaXMuaGFuZGxlU2xvdENoYW5nZX0-PC9zbG90PlxuICAgICAgICA8L3NwYW48PlxuICAgIGA7XG4gIH1cblxuICBwcm90ZWN0ZWQgb3ZlcnJpZGUgZmlyc3RVcGRhdGVkKCkge1xuICAgIHRoaXMuaGFuZGxlU2xvdENoYW5nZSgpO1xuICB9XG4gIFxuICBwcm90ZWN0ZWQgaGFuZGxlU2xvdENoYW5nZSgpIHtcbiAgICB0aGlzLnNsb3RQb3B1bGF0ZWQgPSAhIXRoaXMuaXRlbXMubGVuZ3RoO1xuICB9XG59XG4ifSx7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhRE9DVFlQRSBodG1sPlxuPGhlYWQ-XG4gIDxzY3JpcHQgdHlwZT1cIm1vZHVsZVwiIHNyYz1cIi4vc2xvdC1zbmlmZmVyLmpzXCI-PC9zY3JpcHQ-XG48L2hlYWQ-XG48Ym9keT5cbiAgPHNsb3Qtc25pZmZlcj5cbiAgICA8c3BhbiBzbG90PVwibmFtZWRcIj5pdGVtPC9zcGFuPlxuICA8L3Nsb3Qtc25pZmZlcj5cbjwvYm9keT5cbiJ9LHsibmFtZSI6InBhY2thZ2UuanNvbiIsImNvbnRlbnQiOiJ7XG4gIFwiZGVwZW5kZW5jaWVzXCI6IHtcbiAgICBcImxpdFwiOiBcIl4yLjAuMFwiLFxuICAgIFwiQGxpdC9yZWFjdGl2ZS1lbGVtZW50XCI6IFwiXjEuMC4wXCIsXG4gICAgXCJsaXQtZWxlbWVudFwiOiBcIl4zLjAuMFwiLFxuICAgIFwibGl0LWh0bWxcIjogXCJeMi4wLjBcIlxuICB9XG59IiwiaGlkZGVuIjp0cnVlfV0)
Would this work?